### PR TITLE
Need ClientAwsConfig for the new AWS feature 

### PR DESCRIPTION
**Pull request by ihsandemir**
Introduced the new configuration for the AWS client feature.

### CODE REVIEW

File: hazelcast/include/hazelcast/client/config/ClientAwsConfig.h [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+                /**
+                 * Gets the connection timeout in seconds.
+                 *
+                 * @return the connectionTimeoutSeconds; connection timeout in seconds
+                 * @see #setConnectionTimeoutSeconds(int32_t)
+                 */
+                int32_t getConnectionTimeoutSeconds() const;
+
+                /**
+                 * Sets the connect timeout in seconds. See {@link TcpIpConfig#setConnectionTimeoutSeconds(int32_t)} for more information.
```
Comment (sancar) [inline hazelcast/include/hazelcast/client/config/ClientAwsConfig.h]: Should not this be ? 
ClientConfig#setConnectionTimeoutSeconds(int32_t)

File: hazelcast/src/hazelcast/client/config/ClientNetworkConfig.cpp [hunk 2]
```diff
@@ -42,6 +42,15 @@ namespace hazelcast {
                 this->connectionTimeout = connectionTimeoutInMillis;
                 return *this;
             }
+
+            ClientNetworkConfig &ClientNetworkConfig::setAwsConfig(const ClientAwsConfig &clientAwsConfig) {
```
Comment (sancar) [inline hazelcast/src/hazelcast/client/config/ClientNetworkConfig.cpp]: Should not this part also be guarded by  `HZ_BUILD_WITH_SSL` ?

File: hazelcast/include/hazelcast/client/config/ClientAwsConfig.h (hunk 1 above)
Comment (ihsandemir) [inline hazelcast/include/hazelcast/client/config/ClientAwsConfig.h]: ok, corrected.

File: hazelcast/src/hazelcast/client/config/ClientNetworkConfig.cpp (hunk 2 above)
Comment (ihsandemir) [inline hazelcast/src/hazelcast/client/config/ClientNetworkConfig.cpp]: ok, corrected.

**Pull request merged**